borg/converter.py

```diff
@@ -0,0 +1,150 @@
+from binascii import hexlify
+import os
+
+from .helpers import get_keys_dir
+from .locking import UpgradableLock
+from .repository import Repository, MAGIC
+from .key import KeyfileKey, KeyfileNotFoundError
+
+class AtticRepositoryConverter(Repository):
+    def convert(self, dryrun=True):
+        """convert an attic repository to a borg repository
+
+        those are the files that need to be converted here, from most
+        important to least important: segments, key files, and various
+        caches, the latter being optional, as they will be rebuilt if
+        missing."""
+        print("reading segments from attic repository using borg")
+        segments = [ filename for i, filename in self.io.segment_iterator() ]
+        try:
+            keyfile = self.find_attic_keyfile()
+        except KeyfileNotFoundError:
+            print("no key file found for repository")
+        else:
+            self.convert_keyfiles(keyfile, dryrun)
+        self.close()
+        # partial open: just hold on to the lock
+        self.lock = UpgradableLock(os.path.join(self.path, 'lock'),
+                                   exclusive=True).acquire()
+        self.convert_segments(segments, dryrun)
```

Comment:
put a try/finally here so it releases the lock even if convert_segments crashes.